lib/puppet/ssl/openssl_loader.rb

```diff
@@ -0,0 +1,24 @@
+require 'puppet/util/platform'
+
+# This file should be required instead of writing `require 'openssl'`
+# or any library that loads openssl like `net/https`. This allows the
+# core Puppet code to load correctly in JRuby environments that do not
+# have a functioning openssl (eg a FIPS enabled one).
+
+unless Puppet::Util::Platform.jruby_fips?
+  require 'openssl'
+  require 'net/https'
+else
+  # Even in JRuby we need to define the constants that are wrapped in
+  # Indirections: Puppet::SSL::{Key, Certificate, CertificateRequest}
+  module OpenSSL
+    module PKey
+      class RSA; end
+    end
+
+    module X509
+      class Request; end
+      class Certificate; end
+    end
+  end
```

Comment:
When puppet is required the cert/key indirections are loaded. I'm kind of amazed that works given the empty definitions above. Puppet should no longer call those indirections for anything normally, but I'm curious what would happen if a report processor or puppetdb terminus were to call `Puppet::SSL::Host.localhost`, which will trigger the respective termini. Should we be more defensive about that method and possibly setting the cert/key termini to `:none` or some stupid thing like that? I'm fine if the answer is no, just curious how things might fail.

Reply:
:mindblown:

Reply:
I would assume any call to `Host.localhost` would fail with an undefined constant exception. I'll look into setting the termini to `:none` though I'm not sure I understand why someone would use `.localhost` who wasn't needing to bootstrap Puppet's files.